src/chk/chk_internal.h

```diff
@@ -248,6 +248,15 @@ struct chk_property {
 	uint32_t			cp_rank_nr;
 };
 
+struct chk_iv {
+	uint64_t		 ci_gen;
+	d_rank_t		 ci_rank;
+	uuid_t			 ci_uuid;
+	uint32_t		 ci_phase;
+	uint32_t		 ci_status;
+	uint32_t		 ci_to_leader:1;
```

Comment:
this is a bit strange. I thought you can identify this by different callback().

iv_ent_update() (leaf to leader, it may be called on leader itself, but you can get it by checking the rank? no?).

iv_ent_refresh() (leader to leaf)

Reply:
When leader triggers IV update, it will set ci_to_leader as zero, then chk_iv_ent_update will always return -DER_IVCB_FORWARD for such IV message on all engines. It is expected for IV, right?

Reply:
> When leader triggers IV update, it will set ci_to_leader as zero, then chk_iv_ent_update will always return -DER_IVCB_FORWARD for such IV message on all engines. It is expected for IV, right?


DER_IVCB_FORWARD() is checked in iv_ent_update_cb(), if the master rank != itself rank, then it will return -DER_IVCB_FORWARD, it does not need a special flag to do it.
rebuild_iv_ent_update() ?   Hmm, I thought ns will include the master rank in your case? no?

Reply:
For DAOS check IV, the check leader is the master, the check leader and check engine may reside on the same rank.